Dockerfile

```diff
@@ -1,84 +1,105 @@
+ENV PATH="/root/.local/bin/:$PATH"
+
+RUN apt-get update \
+    && apt-get install -y \
+       curl \
+       git \
+       apt-transport-https \
+       gnupg2 \
+       build-essential \
+       unzip
 
 WORKDIR /app
 
-# zscaler trust - uncomment for building image locally
-#COPY  zscaler.root.crt /usr/local/share/ca-certificates/
-#RUN chmod 644 /usr/local/share/ca-certificates/*.crt && update-ca-certificates
-
-RUN apt-get update && apt-get install -y \
-    curl \
-    git \
-    apt-transport-https \
-    gnupg2 \
-    build-essential \
-    && curl -fsSL https://pkgs.k8s.io/core:/stable:/v1.29/deb/Release.key | gpg --dearmor -o /etc/apt/keyrings/kubernetes-apt-keyring.gpg \
-    && echo 'deb [signed-by=/etc/apt/keyrings/kubernetes-apt-keyring.gpg] https://pkgs.k8s.io/core:/stable:/v1.29/deb/ /' | tee /etc/apt/sources.list.d/kubernetes.list \
-    && apt-get update \
-    && apt-get install -y kubectl unzip\
-    && rm -rf /var/lib/apt/lists/* 
 
-# Install AWS CLI v2 so kubectl works w/ remote eks clusters
-# build-arg to choose architecture of the awscli binary x68_64 or aarch64 - defaulting to x86_64
+# Create and activate virtual environment
+RUN python -m venv /app/venv --upgrade-deps && \
+    . /app/venv/bin/activate
+
+ENV VIRTUAL_ENV=/app/venv
+ENV PATH="$VIRTUAL_ENV/bin:$PATH"
+
+# Needed for kubectl
+RUN curl -fsSL https://pkgs.k8s.io/core:/stable:/v1.29/deb/Release.key -o Release.key
+
+# Set up packages for aws cli
 ARG ARCH=x86_64
 RUN curl "https://awscli.amazonaws.com/awscli-exe-linux-${ARCH}.zip" -o "awscliv2.zip" \
     && unzip awscliv2.zip \
     && ./aws/install
 
-# Install Google cli so kubectl works w/ remove gke clusters
+# Set up packages for gcloud
 RUN curl https://dl.google.com/dl/cloudsdk/release/google-cloud-sdk.tar.gz > /tmp/google-cloud-sdk.tar.gz
 RUN mkdir -p /usr/local/gcloud \
   && tar -C /usr/local/gcloud -xvf /tmp/google-cloud-sdk.tar.gz \
   && /usr/local/gcloud/google-cloud-sdk/install.sh
-ENV PATH $PATH:/usr/local/gcloud/google-cloud-sdk/bin
-RUN gcloud components install gke-gcloud-auth-plugin
-
-# Install Krew and add its installation directory to PATH
-RUN sh -c "\
-    set -x; cd \$(mktemp -d) && \
-    OS=\$(uname | tr '[:upper:]' '[:lower:]') && \
-    ARCH=\$(uname -m | sed -e 's/x86_64/amd64/' -e 's/\\(arm\\)\\(64\\)\\?.*/\\1\\2/' -e 's/aarch64$/arm64/') && \
-    KREW=krew-\${OS}_\${ARCH} && \
-    curl -fsSLO \"https://github.com/kubernetes-sigs/krew/releases/latest/download/\${KREW}.tar.gz\" && \
-    tar zxvf \"\${KREW}.tar.gz\" && \
-    ./\"\${KREW}\" install krew \
-    "
-
-# Add Krew to PATH
-ENV PATH="/root/.krew/bin:$PATH"
 
-# Install kube-lineage via Krew
-RUN kubectl krew install lineage
-
-# Copy the poetry configuration files into the container at /app
-COPY pyproject.toml poetry.lock* /app/
+# Set the architecture-specific kube lineage URLs
+ARG ARM_URL=https://github.com/Avi-Robusta/kube-lineage/releases/download/v2.0.1/kube-lineage-macos-latest-v2.0.1
```

Comment:
Do we still need the aws and gcp CLIs on the builder image?

Reply:
no, ill remove it (but the above urls are kube-lineage)